File: hw2/src/ship.cpp
#include <string_view>
#include "ship.h"

Ship::Ship(int speed, float distance, int displacement, VesselType vesselType)
    : BaseTransport(speed, distance), m_displacement(displacement), m_vesselType(vesselType)
{
}

void Ship::Out(std::ostream& out)
{
    std::string_view vesselTypeStr;
    switch (m_vesselType)
    {
    case VesselType::Liner:
        vesselTypeStr = "liner";
        break;
    case VesselType::Tow:
        vesselTypeStr = "tow";
        break;
    case VesselType::Tanker:
        vesselTypeStr = "tanker";
        break;
    }

    out << "displacement = " << m_displacement << ", vessel type = " << vesselTypeStr << std::endl;
}

std::string_view Ship::GetName()
{
    return "Ship";
}

File: hw3/src/ship.py
'''
ship.py - содержит описание корабля
'''

from enum import Enum
from base_transport import *

# Корабль
class Ship(BaseTransport):
    # Тип судна
    class VesselType(Enum):
        liner = 1 # лайнер
        tow = 2 # буксир
        tanker = 3 # танкер

        def __str__(self):
            return self.name

    # Инициализация
    def __init__(self, speed, distance, displacement, vessel_type):
        super().__init__(speed, distance)
        self.displacement = displacement
        self.vessel_type = vessel_type

    # Конвертация в строку
    def __str__(self):
        return super().__str__() + f', displacement = {self.displacement}, vessel_type = {self.vessel_type}'
